src/storage/src/sink/kafka.rs

```diff
@@ -803,12 +792,22 @@ impl KafkaSinkState {
 
             let mut latest_ts = None;
             let mut latest_offset = None;
-            while let Some((message, offset)) = get_next_message(&mut consumer, timeout)? {
+
+            while let Some((key, message, offset)) = get_next_message(&mut consumer, timeout)? {
                 debug_assert!(offset >= latest_offset.unwrap_or(0));
                 latest_offset = Some(offset);
 
-                if let Some(ts) = maybe_decode_consistency_end_record(&message, consistency_topic)?
-                {
+                let timestamp_opt = {
+                    match std::str::from_utf8(&key) {
+                        Ok(str_key) if str_key == consistency_key => {
```

Comment:
Not sure if you're going to be poking around here with follow-ups.  If yeah, I have a similar nit as another PR about not avoiding wildcard matches.  If not, I think I have some work around here soon

Reply:
No plans but no objection!